src/Contracts/DataCollector.php

```diff
@@ -2,10 +2,19 @@
 
 namespace AG\ElasticApmLaravel\Contracts;
 
+use AG\ElasticApmLaravel\Agent;
 use Illuminate\Support\Collection;
 
 interface DataCollector
 {
+    /*
+     * This allows making a concrete Agent available to the collector after everything
+     * is booted. The JobCollector class makes extensive use of the Agent, unlike other
+     * collectors. I think long term, creating a separate "middleware" for jobs will
```

Comment:
We already provide a [Job Middleware](https://github.com/arkaitzgarro/elastic-apm-laravel/blob/master/src/Jobs/Middleware/RecordTransaction.php#L7), if this is what you mean. The job collector was developed for Laravel 5.5 compatibility.

Reply:
I did see the job middleware. I was reacting more the difference between how HTTP transactions are collected and how jobs are collected. With the HTTP transaction, the collector only registers listeners and the middleware does all of the work. The job collector, on the other hand does all the interaction with the agent and other work. That couples the JobCollector to the Agent in ways that other Collectors are not.

My comment was not clear on that, mostly because my thinking on subject evolved after making it. I've had a chance to learn more of your package now.